TheJumpCloud/support
# Dc 424 replace deprecated ms online cmdlets

## Issues
* [DC-424](https://jumpcloud.atlassian.net/browse/DC-424) - Replace deprecated MSOnline cmdlets

## What does this solve?
Replaces the deprecated MSOnline cmdlets with their MgGraph equivalents

## Is there anything particularly tricky?

## How should this be tested?
Verify that the commands Enable-JumpCloud.Office365.SSO, Disable-JumpCloud.Office365.SSO & Show-JumpCloud.Office365.SSO work correctly

1. Download the Zipped `/Support` repository on [this branch](https://github.com/TheJumpCloud/support/tree/DC-424-Replace-deprecated-MSOnline-cmdlets)
2. Unzip the contents of the downloaded .zip file
3. CD to the `/support/PowerShell/JumpCloud.Office365.SSO` directory
4. On the JumpCloud console, either configure or download the XML metadata for the Office365 SSO connector
5. 
![Screenshot 2024-04-26 at 2 49 13 PM](https://github.com/TheJumpCloud/support/assets/54448601/01ecde35-239d-495a-9344-51832f15a418)
6. Save the Metadata.XML file in the `/support/PowerShell/JumpCloud.Office365.SSO` directory.
7. Import the `Office356.SSO` module: `import-module /Users/jworkman/Documents/GitHub/support/PowerShell/JumpCloud.Office365.SSO/JumpCloud.Office365.SSO.psd1 -Force`
8. Enable federation on the domain and point to the metadata xml file: `enable-JumpCloud.Office365.SSO -XMLFilePath ./JumpCloud-office365-metadata.xml`  you should see a message that the domain was federated.
![Screenshot 2024-04-26 at 2 47 45 PM](https://github.com/TheJumpCloud/support/assets/54448601/1bc4897c-d4a4-4198-ae0f-9fcbce11d11f)
9. Attempt to authenticate as a user to the EntraID domain through the JumpCloud user portal
10. The user should be able to login to EntraID through JumpCloud SSO


## Screenshots


[DC-424]: https://jumpcloud.atlassian.net/browse/DC-424?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
